Completed Assignment 2

ffarashid opened the pull request:
<!-- PULL REQUEST TITLE: UofT-DSI | <Module Name> - Assignment <assignment number>-->

## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)

## What did you learn from the changes you have made?

## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?

## How were these changes tested?

## A reference to a related issue in your repository (if applicable)

## Checklist
- [ ] I can confirm that my changes are working as intended

ffarashid committed:
Completed Assignment 2

anjali-deshpande-hub reviewed (changes requested):
Well done! Just one change requested: 
Replace the hardcoded absolute paths to relative paths so that a Learning support can run the code successfully as well. For example: change `r"C:\Users\farha\python\05_src\data\assignment_2_data\inflammation_01.csv"` to `r"..\..\05_src\data\assignment_2_data\inflammation_01.csv"`

ffarashid committed:
Revised assignment_2.ipynb

ffarashid committed:
Update_02 assignment_2.ipynb

anjali-deshpande-hub reviewed (changes requested):
Review 2:
The assignment_2.ipynb is giving me an 'Invalid Notebook' error. Could you please check? [https://github.com/ffarashid/python/blob/assignment-2/02_activities/assignments/assignment_2.ipynb](https://github.com/ffarashid/python/blob/assignment-2/02_activities/assignments/assignment_2.ipynb)
Screenshot attached:
![Screenshot 2025-05-20 170315](https://github.com/user-attachments/assets/5125a6b3-caca-4b6b-8679-17b063177983)

ffarashid commented:
Hi,

Please check this pull request:

https://github.com/ffarashid/python/pull/2/commits
[https://opengraph.githubassets.com/c27bd37c562b8846a863122bf920ddd5cd5166ff5bf4280aedb953a282faea7c/ffarashid/python/pull/2]<https://github.com/ffarashid/python/pull/2/commits>
Completed Assignment 2 by ffarashid · Pull Request #2 · ffarashid/python<https://github.com/ffarashid/python/pull/2/commits>
What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports) What did you learn from the changes you have made? Was there another approach you were...
github.com

Best regards,
Farhan Rashid
________________________________
From: anjali-deshpande-hub ***@***.***>
Sent: Tuesday, May 20, 2025 5:04 PM
To: ffarashid/python ***@***.***>
Cc: Farhan Rashid ***@***.***>; Author ***@***.***>
Subject: Re: [ffarashid/python] Completed Assignment 2 (PR #2)


@anjali-deshpande-hub requested changes on this pull request.

Review 2:
The assignment_2.ipynb is giving me an 'Invalid Notebook' error. Could you please check? https://github.com/ffarashid/python/blob/assignment-2/02_activities/assignments/assignment_2.ipynb
Screenshot attached:
Screenshot.2025-05-20.170315.png (view on web)<https://github.com/user-attachments/assets/5125a6b3-caca-4b6b-8679-17b063177983>

—
Reply to this email directly, view it on GitHub<https://github.com/ffarashid/python/pull/2#pullrequestreview-2855589248>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AOXYUMHKXQLGTVVVLECKVML27OKFJAVCNFSM6AAAAAB44C2CQOVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDQNJVGU4DSMRUHA>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

ffarashid committed:
Updated May 20th, 5:34 pm

anjali-deshpande-hub commented:
Hello Farhan!

I am reviewing your latest commit. If you download/load the assignment_2.ipynb file in browser or VSCode, it displays the 'Invalid Notebook" error. 
I think its because of an invalid string. If you notice carefully, all backslashes are not escaped.  It should be:   "r\"..\\..\\05_src\\data\\assignment_2_data\\inflammation_01.csv\",\n" instead of "r\"..\..\\05_src\\data\\assignment_2_data\\inflammation_01.csv\",\n"
![Screenshot 2025-05-20 174004](https://github.com/user-attachments/assets/08cf2e3d-88c2-40e4-8b0a-7f934a2656f1)

Hope this helps!

ffarashid committed:
Merge branch 'assignment-2' of https://github.com/ffarashid/python into assignment-2

ffarashid commented:
Can you please check this commit: https://github.com/ffarashid/python/pull/2/commits/c9d1508cac5a3c87aef7e88b6cc263ae6762f223

Best regards,
Farhan Rashid
________________________________
From: anjali-deshpande-hub ***@***.***>
Sent: Tuesday, May 20, 2025 5:43 PM
To: ffarashid/python ***@***.***>
Cc: Farhan Rashid ***@***.***>; Author ***@***.***>
Subject: Re: [ffarashid/python] Completed Assignment 2 (PR #2)

[https://avatars.githubusercontent.com/u/167125913?s=20&v=4]anjali-deshpande-hub left a comment (ffarashid/python#2)<https://github.com/ffarashid/python/pull/2#issuecomment-2895901434>

Hello Farhan!

I am reviewing your latest commit. If you download/load the assignment_2.ipynb file in browser or VSCode, it displays the 'Invalid Notebook" error.
I think its because of an invalid string. If you notice carefully, all backslashes are not escaped. It should be: "r"..\..\05_src\data\assignment_2_data\inflammation_01.csv",\n" instead of "r"....\05_src\data\assignment_2_data\inflammation_01.csv",\n"
Screenshot.2025-05-20.174004.png (view on web)<https://github.com/user-attachments/assets/08cf2e3d-88c2-40e4-8b0a-7f934a2656f1>

Hope this helps!

—
Reply to this email directly, view it on GitHub<https://github.com/ffarashid/python/pull/2#issuecomment-2895901434>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AOXYUMCFZD4CYXFTK43XEYL27OOYTAVCNFSM6AAAAAB44C2CQOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDQOJVHEYDCNBTGQ>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

anjali-deshpande-hub commented:
I can load the file successfully now. 
But the hardcoded path still exists. If I run the code on my machine, it will give me error. In other words, the code is not portable, Therefore to fix it, please replace file paths
r"C:\Users\farha\python\05_src\data\assignment_2_data\inflammation_01.csv" 
 to 
r"..\..\05_src\data\assignment_2_data\inflammation_01.csv"

I hope you understand the reason behind the change.

ffarashid committed:
Updated assignment_2.ipynb

ffarashid commented:
The reason for it is just for you to test the code at your end, and your directory path is different than mine, right? Can you check it now?


[cid:997e5a3b-dbc1-49b1-9ce5-7d3570aa7e4e]

Best regards,
Farhan Rashid
________________________________
From: anjali-deshpande-hub ***@***.***>
Sent: Tuesday, May 20, 2025 7:15 PM
To: ffarashid/python ***@***.***>
Cc: Farhan Rashid ***@***.***>; Author ***@***.***>
Subject: Re: [ffarashid/python] Completed Assignment 2 (PR #2)

[https://avatars.githubusercontent.com/u/167125913?s=20&v=4]anjali-deshpande-hub left a comment (ffarashid/python#2)<https://github.com/ffarashid/python/pull/2#issuecomment-2896037654>

I can load the file successfully now.
But the hardcoded path still exists. If I run the code on my machine, it will give me error. In other words, the code is not portable, Therefore to fix it, please replace file paths
r"C:\Users\farha\python\05_src\data\assignment_2_data\inflammation_01.csv"
to
r"....\05_src\data\assignment_2_data\inflammation_01.csv"

I hope you understand the reason behind the change.

—
Reply to this email directly, view it on GitHub<https://github.com/ffarashid/python/pull/2#issuecomment-2896037654>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AOXYUMBQTZZ6W6LNK4U45BL27OZPPAVCNFSM6AAAAAB44C2CQOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDQOJWGAZTONRVGQ>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

anjali-deshpande-hub reviewed (changes requested):
The submission is still incorrect — the \ is missing.

Please make sure to run the code locally on your machine before submitting. The code should have given error on your machine.
Verifying locally will help avoid repeated issues and save time for reviewing.

ffarashid commented:
Hi,

I have replaced with exactly what you asked: [cid:0ff94f28-0647-4d05-b5bd-0b83a042531d]

I have been receiving the error since I removed the directory path to "...."

May I am not understanding the significance. Can you explain or maybe refer me to handouts?

Best regards,
Farhan Rashid
________________________________
From: anjali-deshpande-hub ***@***.***>
Sent: Tuesday, May 20, 2025 7:40 PM
To: ffarashid/python ***@***.***>
Cc: Farhan Rashid ***@***.***>; Author ***@***.***>
Subject: Re: [ffarashid/python] Completed Assignment 2 (PR #2)


@anjali-deshpande-hub requested changes on this pull request.

The submission is still incorrect — the \ is missing.

Please make sure to run the code locally on your machine before submitting. The code should have given error on your machine.
Verifying locally will help avoid repeated issues and save time for reviewing.

—
Reply to this email directly, view it on GitHub<https://github.com/ffarashid/python/pull/2#pullrequestreview-2855827763>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AOXYUMER7M5XG4X2HMRXF5T27O4QTAVCNFSM6AAAAAB44C2CQOVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDQNJVHAZDONZWGM>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

ffarashid committed:
Update assignment_2.ipynb

ffarashid commented:
Now it worked when I added /

https://github.com/ffarashid/python/pull/2/commits

Best regards,
Farhan Rashid
________________________________
From: Farhan Feroz Ali Rashid ***@***.***>
Sent: Tuesday, May 20, 2025 7:48 PM
To: ffarashid/python ***@***.***>; ffarashid/python ***@***.***>
Cc: Author ***@***.***>
Subject: Re: [ffarashid/python] Completed Assignment 2 (PR #2)

Hi,

I have replaced with exactly what you asked: [cid:0ff94f28-0647-4d05-b5bd-0b83a042531d]

I have been receiving the error since I removed the directory path to "...."

May I am not understanding the significance. Can you explain or maybe refer me to handouts?

Best regards,
Farhan Rashid
________________________________
From: anjali-deshpande-hub ***@***.***>
Sent: Tuesday, May 20, 2025 7:40 PM
To: ffarashid/python ***@***.***>
Cc: Farhan Rashid ***@***.***>; Author ***@***.***>
Subject: Re: [ffarashid/python] Completed Assignment 2 (PR #2)


@anjali-deshpande-hub requested changes on this pull request.

The submission is still incorrect — the \ is missing.

Please make sure to run the code locally on your machine before submitting. The code should have given error on your machine.
Verifying locally will help avoid repeated issues and save time for reviewing.

—
Reply to this email directly, view it on GitHub<https://github.com/ffarashid/python/pull/2#pullrequestreview-2855827763>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AOXYUMER7M5XG4X2HMRXF5T27O4QTAVCNFSM6AAAAAB44C2CQOVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDQNJVHAZDONZWGM>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

ffarashid committed:
Update assignment_2.ipynb

anjali-deshpande-hub reviewed (approved):
Fixed finally!! Good effort!
